### DISCUSSION TITLE
To show the desktop I am on

### DISCUSSION THREAD

**Discussion by KVolker** [FuPeiJiang/VD.ahk#23] [Show and tell]
I like to have a sign of which desktop I see at the moment. I tried to change the tray icon of the actual running script. Like this.
```
ActiveDesktopGetNumber(){
    v := new VD()
    v.init()
    n := v.getCurrentDesktopNum()
    v := ""
    return n
}

ActiveDesktopSetIcon(){
    n := ActiveDesktopGetNumber()
    w =  %A_ScriptDir%
    if (n<10){
        f := w "\icons\ico\Number-" n ".ico"
    }
    else {
        f := w "\icons\ico\Math-greater-than.ico"
    }
    IfExist, %f%
	    Menu, Tray, Icon, %f% 
}
```
I also set up a timer to call every second this function as I might change the virtual desktop not only by my script but also by all the other methods Windows offers for that.

I use [these icons from IconArchive](https://iconarchive.com/show/red-orb-alphabet-icons-by-iconarchive.1.html).

You should right-click on the taskbar and then in settings choose the AHK icon to be visible in the tray, not hidden in the display you get by clicking ">".

**Comment by micjoh** [comment]
Just drag the icon to the visible are to the right of the ^ sign. Works for me
